src/main/java/at/hannibal2/skyhanni/features/inventory/experimentationtable/SuperpairDataDisplay.kt

```diff
@@ -0,0 +1,257 @@
+import at.hannibal2.skyhanni.events.GuiContainerEvent.SlotClickEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.DelayedRun
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.milliseconds
+
+@SkyHanniModule
+// TODO important: all use cases of listOf in combination with string needs to be gone. no caching, constant new list creation, and bad design.
+object SuperpairDataDisplay {
+
+    private val config get() = SkyHanniMod.feature.inventory.experimentationTable
+
+    data class Item(val index: Int, val reward: String, val damage: Int)
+    data class ItemPair(val first: Item, val second: Item)
+    data class FoundData(val item: Item = emptyItem, val pair: ItemPair = emptyPair)
+
+    private val emptyItem = Item(-1, "", -1)
+    private val emptyPair = ItemPair(emptyItem, emptyItem)
+
+    private enum class FoundType {
+        NORMAL,
+        POWERUP,
+        MATCH,
+        PAIR
+        ;
+
+        fun isOneOf(vararg types: FoundType) = types.contains(this)
+    }
+
+    private val sideSpaces1 = listOf(17, 18, 26, 27, 35, 36)
+    private val sideSpaces2 = listOf(16, 17, 18, 19, 25, 26, 27, 28, 34, 35, 36, 37)
+
+    private var display = emptyList<String>()
+    private var uncoveredItems = mutableMapOf<Int, Item>()
+    private var found = mutableMapOf<FoundData, FoundType>()
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        display = emptyList()
+
+        uncoveredItems = mutableMapOf()
+        found.clear()
+    }
+
+    @SubscribeEvent
+    fun onChestGuiOverlayRendered(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (display.isEmpty()) display = drawDisplay()
+
+        config.superpairDisplayPosition.renderStrings(display, posLabel = "Superpair Experiment Information")
+    }
+
+    @SubscribeEvent
+    fun onSlotClick(event: SlotClickEvent) {
+        if (!isEnabled()) return
+        val currentExperiment = ExperimentationTableAPI.getCurrentExperiment() ?: return
+
+        val item = event.item ?: return
+        if (isOutOfBounds(event.slotId, currentExperiment) || item.displayName.removeColor() == "?") return
+
+        val clicksItem = InventoryUtils.getItemAtSlotIndex(4)
+
+        if (uncoveredItems.none { it.value.index == event.slotId && it.key == uncoveredItems.keys.max() }) {
+            if (clicksItem != null && clicksItem.displayName.removeColor().split(" ")[1] == "0") return
```

Comment:
the display name here should either be replaced by internal name check instead, or leta data check, or check name aginst repo pattern